nifi-nar-bundles/nifi-adx-bundle/nifi-adx-processors/src/main/java/org/apache/nifi/processors/adx/AzureAdxSourceProcessor.java

```diff
@@ -0,0 +1,192 @@
+            .displayName(AzureAdxSourceProcessorParamsEnum.DB_NAME.getParamDisplayName())
+            .description(AzureAdxSourceProcessorParamsEnum.DB_NAME.getParamDescription())
+            .required(true)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .build();
+    public static final PropertyDescriptor ADX_QUERY = new PropertyDescriptor
+            .Builder().name(AzureAdxSourceProcessorParamsEnum.ADX_QUERY.name())
+            .displayName(AzureAdxSourceProcessorParamsEnum.ADX_QUERY.getParamDisplayName())
+            .description(AzureAdxSourceProcessorParamsEnum.ADX_QUERY.getParamDescription())
+            .required(true)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .build();
+    public static final PropertyDescriptor ADX_SOURCE_SERVICE = new PropertyDescriptor
+            .Builder().name(AzureAdxSourceProcessorParamsEnum.ADX_SOURCE_SERVICE.name())
+            .displayName(AzureAdxSourceProcessorParamsEnum.ADX_SOURCE_SERVICE.getParamDisplayName())
+            .description(AzureAdxSourceProcessorParamsEnum.ADX_SOURCE_SERVICE.getParamDescription())
+            .required(true)
+            .identifiesControllerService(AdxSourceConnectionService.class)
+            .build();
+    private final ObjectMapper objectMapper = new ObjectMapper();
+    private Set<Relationship> relationships;
+    private List<PropertyDescriptor> descriptors;
+    private Client executionClient;
+
+    @Override
+    protected void init(final ProcessorInitializationContext context) {
+        this.descriptors = List.of(ADX_SOURCE_SERVICE,DB_NAME,ADX_QUERY);
+        this.relationships = Set.of(RL_SUCCEEDED,RL_FAILED);
+    }
+
+    @Override
+    public Set<Relationship> getRelationships() {
+        return this.relationships;
+    }
+
+    @Override
+    public final List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return descriptors;
+    }
+
+    @OnScheduled
+    public void onScheduled(final ProcessContext context) {
+        AdxSourceConnectionService service = context.getProperty(ADX_SOURCE_SERVICE).asControllerService(AdxSourceConnectionService.class);
+        executionClient = service.getKustoQueryClient();
+    }
+
+    @Override
+    public void onTrigger(ProcessContext context, ProcessSession session) throws ProcessException {
+        FlowFile outgoingFlowFile;
+        String databaseName = context.getProperty(DB_NAME).getValue();
+        String adxQuery;
+        KustoOperationResult kustoOperationResult;
+
+        //checks if this processor has any preceding connection, if yes retrieve
+        if ( context.hasIncomingConnection() ) {
+            FlowFile incomingFlowFile = session.get();
+            //incoming connection exists but the incoming flowfile is null
+            if ( incomingFlowFile == null && context.hasNonLoopConnection() ) {
+                getLogger().error("Incoming connection exists but Incoming flow file is null");
```

Comment:
This status should not be logged as an error, it is a framework concern, so recommend changing the log to an info or removing the log.